src/utils/frame.js

```diff
@@ -160,6 +171,48 @@ export function annotateFrame(frame: Frame) {
   return frame;
 }
 
+function annotateBabelAsyncFrames(frames: Frame[]) {
+  const startIndexes = frames.reduce(
+    (accumulator, frame, index) =>
+      isStartOfBabelGroup(frame) ? [...accumulator, index] : accumulator,
+    []
+  );
+  const endIndexes = frames.reduce(
+    (accumulator, frame, index) =>
+      isEndOfBabelGroup(frame) ? [...accumulator, index] : accumulator,
+    []
+  );
```

Comment:
can we generalize this so we have a new `getIndices` function that takes frames, and a predicate?

Reply:
absolutely.